Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -461,7 +461,15 @@ class SettingsTableViewController: UITableViewController {
     @objc private func SELrefresh() {
         // Through-out, be aware that modifying the control while a refresh is in progress is /not/ supported and will likely crash the app.
         if let account = self.profile.getAccount() {
-            account.advance().upon { _ in
+            account.advance().upon { state in
```

Comment:
I notice we have the same code in the BVC. Does it make sense to move this logic into an FxAState? My understanding of how the FxAState works is that it goes step by step through the different authentication states serially. I could also be totally wrong - @ncalexan would know best :)